plantcv/plantcv/roi_objects.py

```diff
@@ -82,6 +83,71 @@ def roi_objects(img, roi_type, roi_contour, roi_hierarchy, object_contour, obj_h
         cv2.drawContours(ori_img, kept_cnt, -1, (0, 255, 0), -1, lineType=8, hierarchy=hierarchy)
         cv2.drawContours(ori_img, roi_contour, -1, (255, 0, 0), 5, lineType=8, hierarchy=roi_hierarchy)
 
+    # Find the largest contour if roi_type is set to 'largest'
+    elif roi_type == 'largest':
+        # Print warning statement about this feature
+        print(
+            "Warning: roi_type='largest' will only return the largest extreme outer contour. All child contours are left behind.")
+
+        # Filter contours outside of the region of interest
+        for c, cnt in enumerate(object_contour):
+            length = (len(cnt) - 1)
+            stack = np.vstack(cnt)
+            keep = False
+            # Test if the contours are within the ROI
+            for i in range(0, length):
+                pptest = cv2.pointPolygonTest(roi_contour[0], (stack[i][0], stack[i][1]), False)
+                if int(pptest) != -1:
+                    keep = True
+            if keep:
+                # Color the "gap contours" white
+                if obj_hierarchy[0][c][3] > -1:
+                    cv2.drawContours(w_back, object_contour, c, (255, 255, 255), -1, lineType=8,
+                                     hierarchy=obj_hierarchy)
+                # Color the plant contour parts black
+                else:
+                    cv2.drawContours(w_back, object_contour, c, (0, 0, 0), -1, lineType=8, hierarchy=obj_hierarchy)
+            # If the contour isn't overlapping with the ROI, color it white
+            else:
+                cv2.drawContours(w_back, object_contour, c, (255, 255, 255), -1, lineType=8, hierarchy=obj_hierarchy)
+
+        # Draw the contours kept after filtering by ROI
+        kept = cv2.cvtColor(w_back, cv2.COLOR_RGB2GRAY)
+        kept_obj = cv2.bitwise_not(kept)
+        kept_cnt, hierarchy = cv2.findContours(kept_obj, cv2.RETR_TREE, cv2.CHAIN_APPROX_NONE)[-2:]
+
+        # Find the index from the largest contour in the list of contours
+        largest_area = 0
+        for c, cnt in enumerate(kept_cnt):
+            area = cv2.contourArea(cnt)
```

Comment:
This looks great! I will test it out later today. A future feature that would be helpful could be to create a filter based on contour area. Instead of filling, eroding, dilating, bluring etc to remove noise, you could just drop small contours. So it would make sense to factor this largest area piece into a new, more general function. Just a thought!

Reply:
Would `pcv.fill` work? It evaluates the individual contours by area and removes them if they are smaller than the threshold.

Reply:
Yes Now that I looked at the code I think it does the same thing I was thinking about. I thought `pcv.fill` would also fill holes but it doesn't appear to do so. In fact, there is a `remove_small_holes` in `misc.py` that doesn't appear to be used anywhere?

Reply:
What I’ve done before to fill holes is pcv.invert and then pcv.fill.